stock_cycle_count/models/stock_quant.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2024 ForgeFlow S.L.
+#   (http://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+from odoo import models
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    def _apply_inventory(self):
+        accuracy_dict = {}
+        theoretical_dict = {}
+        counted_dict = {}
+        for rec in self:
+            if rec.discrepancy_percent > 100:
+                line_accuracy = 0
+            else:
+                line_accuracy = 1 - (rec.discrepancy_percent / 100)
+            accuracy_dict[rec.id] = line_accuracy
+            theoretical_dict[rec.id] = rec.quantity
+            counted_dict[rec.id] = rec.inventory_quantity
+        res = super()._apply_inventory()
+        for rec in self:
+            record_moves = self.env["stock.move.line"]
+            moves = record_moves.search(
+                [
+                    ("product_id", "=", rec.product_id.id),
+                    ("lot_id", "=", rec.lot_id.id),
+                    "|",
+                    ("location_id", "=", rec.location_id.id),
+                    ("location_dest_id", "=", rec.location_id.id),
+                ],
+                order="create_date asc",
+            ).filtered(
+                lambda x: not x.company_id.id
```

Comment:
Doesn't it correspond to normal record rule ?

Reply:
In other words, is the `filtered` necessary ?

Reply:
@rousseldenis You are correct, since we are searching for stock move lines by location, and the locations already belong to a company, it is not needed this filter. I'll make the change.

Thanks

Reply:
@rousseldenis @ArnauCForgeFlow  it is not the normal record rule, it is filtering based on the quant company, the normal record rules filters based on the users companies, we cannot rely on the user having the proper companies.

That been said, I think instead of filtering, it can be done with a domain that includes a couple of company_id clauses.